mqtt/mqtt-edgehub/src/command/mod.rs

```diff
@@ -0,0 +1,46 @@
+mod authorized_identities;
+mod disconnect;
+mod handler;
+
+pub use authorized_identities::AuthorizedIdentities;
+pub use disconnect::Disconnect;
+pub use handler::{CommandHandler, ShutdownHandle};
+
+use std::error::Error as StdError;
+
+use mqtt3::ReceivedPublication;
+
+pub trait Command {
+    type Error;
+
+    fn topic(&self) -> &str;
+
+    fn handle(&mut self, publication: &ReceivedPublication) -> Result<(), Self::Error>;
+}
+
+pub struct DynCommand<C> {
```

Comment:
```
/// Wrapper to take any command and wrap error type as a trait object.
///
/// We need this to collect all commands in a `HashMap<dyn Command<Error = Box<dyn StdError + Send>>`.
/// Otherwise there is no good way to bring different commands to the same trait object type with associated trait object error type.
```